packages/muelu/CMakeLists.txt

```diff
@@ -57,6 +57,19 @@ ENDIF()
 TRIBITS_ADD_DEBUG_OPTION()
 TRIBITS_ADD_SHOW_DEPRECATED_WARNINGS_OPTION()
 
+TRIBITS_ADD_OPTION_AND_DEFINE(
+  ${PACKAGE_NAME}_ENABLE_DEPRECATED_CODE
+  HAVE_${PACKAGE_NAME_UC}_DEPRECATED_CODE
```

Comment:
Thanks for adding this!  Now might be the time to think about making this consistent with Kokkos and Tpetra, and also putting the package name first: `MUELU_ENABLE_DEPRECATED_CODE`.

Reply:
@mhoemmen @jhux2 
I do not know what the historical reason is for choosing `HAVE_${PACKAGE_NAME_UC}_DEPRECATED_CODE` over the Tpetra convention but it is consistent in MueLu at this point so unless we make the decision to rename all our preprocessor variables it would not make sense to change just this one...

@mhoemmen maybe a GitHub issue can be opened to discuss why MueLu uses this naming scheme. @cgcgcg hypothesized that it creates a distinction between CMake variables and actual preprocessor variables.

Reply:
I doubt there was a reason other than "that's what Trilinos always did."  MueLu has never bothered much with consistency with other Trilinos packages (see e.g., early debates about method name capitalization) so it's all good ;-)

Reply:
> I do not know what the historical reason is for choosing `HAVE_${PACKAGE_NAME_UC}_DEPRECATED_CODE` over the Tpetra convention but it is consistent in MueLu at this point so unless we make the decision to rename all our preprocessor variables it would not make sense to change just this one...

@lucbv Afaik, this is consistent with what is done throughout Trilinos (see Belos, Epetra, Stokhos, ...).   My understanding is that cmake variables/options start with the package name, preprocessor macros start with `HAVE`.